### DISCUSSION TITLE
Autogenerated link groups support for the <StarlightPage /> component

### DISCUSSION THREAD

**Discussion by mmvarma-pro** [withastro/starlight#1683] [Feature Requests]
### What version of `starlight` are you using?

0.21.2

### What is your idea?

The concept involves organizing collections according to the user's role. For instance, User 1 will exclusively access "Collection 1", while User 2 will be granted access to "Collection 2". User 3, on the other hand, will have the privilege of accessing both collections. The necessary authentication and user role information is already integrated within the Astro page.

### Why is this feature necessary?

We've developed a website dedicated to hosting documentation for all of our products. However, since customers may only subscribe to select products, the site's primary goal is to ensure users access documentation exclusively for the products they have subscribed to. 

Achieving this likely involves additional considerations. But I believe this feature can be a start in that journey. We would greatly appreciate any assistance or insights in this area including any other suggestions as well.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Yes, I agree this would be a helpful extension of the `<StarlightPage />` behaviour and did come up in discussions when adding the component — we decided it could be something to tackle later (that original PR was big and complex enough already 😁), but I think this could be a good way to make usage more closely aligned with the `sidebar` option in `astro.config.mjs`.

**Comment by HiDeoo** [comment]
Starting with [Starlight 0.26.0](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.26.0), the `<StarlightPage />` component `sidebar` prop now accepts an array of [`SidebarItems`](https://starlight.astro.build/reference/configuration/#sidebaritem) like the main Starlight `sidebar` configuration in `astro.config.mjs`.

This change simplifies the definition of sidebar items in the `<StarlightPage />` component and also enables the usage of autogenerated sidebar groups with the `<StarlightPage />` component.
